nixos/modules/services/security/crowdsec.nix

```diff
@@ -0,0 +1,841 @@
+          cfg.settings.lapi.credentialsFile
+        else
+          format.generate "local_api_credentials.yaml" cfg.settings.lapi.credentials;
+      capiFile =
+        if cfg.settings.capi.credentialsFile != null then
+          cfg.settings.capi.credentialsFile
+        else
+          (
+            if cfg.settings.capi.credentials != null then
+              format.generate "central_api_credentials.yaml" cfg.settings.capi.credentials
+            else
+              null
+          );
+
+      tokenFile =
+        if cfg.settings.console.tokenFile != null then
+          cfg.settings.console.tokenFile
+        else
+          (
+            if cfg.settings.console.token != null then
+              pkgs.writeText "console_token.txt" cfg.settings.console.token
+            else
+              null
+          );
+      simulationFile = format.generate "simulation.yaml" cfg.settings.simulation;
+      consoleFile = format.generate "console.yaml" cfg.settings.console.configuration;
+      patternsDir = pkgs.buildPackages.symlinkJoin {
+        name = "crowdsec-patterns";
+        paths = [
+          cfg.localConfig.patterns
+          (lib.attrsets.getOutput "patterns" pkg)
+        ];
+      };
+
+      pkg = cfg.package;
+
+      cscli = pkgs.writeScriptBin "cscli" ''
+        #!${pkgs.runtimeShell}
+        set -eu
+        set -o pipefail
+
+        # cscli needs crowdsec on it's path in order to be able to run `cscli explain`
+        export PATH=$PATH:${lib.makeBinPath [ pkg ]}
+
+        exec ${pkg}/bin/cscli -c=${configFile} "''${@}"
+      '';
+
+      localScenariosMap = (map (format.generate "scenario.yaml") cfg.localConfig.scenarios);
+      localParsersS00RawMap = (
+        map (format.generate "parsers-s00-raw.yaml") cfg.localConfig.parsers.s00Raw
+      );
+      localParsersS01ParseMap = (
+        map (format.generate "parsers-s01-parse.yaml") cfg.localConfig.parsers.s01Parse
+      );
+      localParsersS02EnrichMap = (
+        map (format.generate "parsers-s02-enrich.yaml") cfg.localConfig.parsers.s02Enrich
+      );
+      localPostOverflowsS01WhitelistMap = (
+        map (format.generate "postoverflows-s01-whitelist.yaml") cfg.localConfig.postOverflows.s01Whitelist
+      );
+      localContextsMap = (map (format.generate "context.yaml") cfg.localConfig.contexts);
+      localNotificationsMap = (map (format.generate "notification.yaml") cfg.localConfig.notifications);
+      localProfilesFile = pkgs.writeText "local_profiles.yaml" (
+        lib.strings.concatMapStringsSep "\n---\n" (filename: builtins.readFile filename) (
+          map (prof: format.generate "profile.yaml" prof) cfg.localConfig.profiles
+        )
+      );
+      localAcquisisionFile = pkgs.writeText "local_acquisisions.yaml" (
+        lib.strings.concatMapStringsSep "\n---\n" (filename: builtins.readFile filename) (
+          map (prof: format.generate "acquisition.yaml" prof) cfg.localConfig.acquisitions
```

Comment:
This generates a ton of derivations and does IFD. If we use `builtins.toJSON` it's still compatible and significantly more performant.